pkg/prometheus/operator.go

```diff
@@ -1293,15 +1295,18 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		sErr, ok := err.(*apierrors.StatusError)
 
 		if ok && sErr.ErrStatus.Code == 422 && sErr.ErrStatus.Reason == metav1.StatusReasonInvalid {
-			c.metrics.StsDeleteCreateCounter().Inc()
-
+			if current != nil && current.DeletionTimestamp != nil {
+				level.Info(logger).Log("msg", "updating StatefulSet failed but the resource has already been marked for deletion")
+				return nil
```

Comment:
We should continue rather than return
```suggestion
				continue
```